sqlmesh/core/dialect.py

```diff
@@ -611,12 +613,36 @@ def parse(self: Parser) -> t.Optional[exp.Expression]:
             else:
                 value = self._parse_bracket(self._parse_field(any_token=True))
 
+                if key == "gateway":
+                    gateway = value
+                elif key == "blueprints":
+                    if isinstance(value, exp.Paren):
+                        blueprints = [value.this]
+                    elif isinstance(value, (exp.Tuple, exp.Array)):
+                        blueprints = value.expressions
+                    else:
+                        raise ConfigError(
+                            "The 'blueprints' values need to be enclosed in "
+                            f"parentheses or brackets, got {value} instead."
+                        )
+
+                    # We don't want to include blueprints in the property list
+                    continue
+
             if isinstance(value, exp.Expression):
                 value.meta["sql"] = self._find_sql(start, self._prev)
 
             expressions.append(self.expression(exp.Property, this=key, value=value))
 
-        return self.expression(parser_type, expressions=expressions)
+        expression = self.expression(expression_type, expressions=expressions)
+
+        # We store these properties in the meta to provide quick access at load time
+        if blueprints:
+            expression.meta["blueprints"] = blueprints
```

Comment:
Not sure I like using the meta for this. It's pretty non-obvious and implicit. Plus it introduces even more coupling between sql parsing and model definition in a pretty opaque way. I suggest handling this in the `load_sql_based_model` instead.